src/cow-react/modules/tokens/hooks/useOnchainBalances.ts

```diff
@@ -0,0 +1,144 @@
+import { Interface } from '@ethersproject/abi'
+import { isAddress } from '@src/utils'
+import { CurrencyAmount, Token } from '@uniswap/sdk-core'
+import JSBI from 'jsbi'
+import ERC20ABI from 'abis/erc20.json'
+import { Erc20Interface } from 'abis/types/Erc20'
+import { useMultipleContractSingleData } from 'lib/hooks/multicall'
+import { ListenerOptionsWithGas } from '@uniswap/redux-multicall'
+import { useMemo } from 'react'
+import { OnchainState } from '../types'
+
+const ERC20Interface = new Interface(ERC20ABI) as Erc20Interface
+const DEFAULT_LISTENER_OPTIONS: ListenerOptionsWithGas = { gasRequired: 185_000, blocksPerFetch: 5 }
+
+export type OnchainTokenAmount = OnchainState<CurrencyAmount<Token> | undefined>
+
+export type OnchainTokenAmounts = {
+  [tokenAddress: string]: OnchainTokenAmount
+}
+
+export type OnchainAmountsResult = {
+  amounts: OnchainTokenAmounts
+  isLoading: boolean
+}
+
+export interface OnchainAmountsParams {
+  account?: string
```

Comment:
Can we get rid of optionality here?
Would be cool to leave the responsibility of checking `account` and `tokens` falseness outside of the hook.
I mean, the hook should "think" that `account` and `tokens` always truthy

Reply:
I will put it in the TODO list to review if we want to do it. For now, I kept a lot the compatibility since this is used a lot. 

I think is probably fine. This was used in some place to implement this logic:
- If the orders table is in OPEN tabs
- Then pass undefined in the account, so the hook will not return any tokens balance

I changed this so we pass an empty array of tokens instead. So probably we can do these in all the places we use the hook, but it needs to be reviewed after all refactors